tests/simulation/logical_patch_state_simulation_test.py

```diff
@@ -163,3 +168,99 @@ def test_pauli_product_measurement_distribution(
                 actual_outcome.corresponding_eigenvalue
             )
             assert desired_outcome_probability == pytest.approx(actual_outcome_probability)
+
+
+@pytest.mark.parametrize(
+    "assoc_data_prob",
+    [
+        [("a", 0.50), ("b", 0.30), ("c", 0.20)],
+        [
+            ("a", 1.0),
+            ("b", 0),
+            ("c", 0),
+        ],
+    ],
+)
+def test_proportional_choice(assoc_data_prob):
```

Comment:
what does this function do? The tests look good, but I would like what this function is trying to achieve

Reply:
Internal use. Noted in comment